src/vivarium/framework/population/manager.py

```diff
@@ -389,7 +389,7 @@ def register_simulant_initializer(
         self,
         component: Component | Manager,
         creates_columns: str | Sequence[str] = (),
-        required_resources: Iterable[str | Resource] = (),
+        required_resources: Sequence[str | Resource] = (),
```

Comment:
Why did we move from Iterable to Sequence? It seems to me that Iterable was fine since we don't care about the ordering.